src/store/utils/HistoryItem/types.js

```diff
@@ -0,0 +1,71 @@
+// @flow strict
+
+export const TRANSFER_IN_TYPE: '$TransferIn' = '$TransferIn'
+export const TRANSFER_OUT_TYPE: '$TransferOut' = '$TransferOut'
+export const TRANSFER_CANCEL_TYPE: '$TransferCancel' = '$TransferCancel'
+export const CONTRACT_CALL_TYPE: '$ContractCall' = '$ContractCall'
+export const EVENT_MINT_TYPE: '$EventMint' = '$EventMint'
+export const EVENT_BURN_TYPE: '$EventBurn' = '$EventBurn'
+
+export type HistoryItemsTypes =
+  '$TransferIn'
+  | '$TransferOut'
+  | '$TransferCancel'
+  | '$ContractCall'
+  | '$EventMint'
+  | '$EventBurn'
+
+export type TransactionState
+  = 'success'
+  | 'pending'
+  | 'fail'
+  | 'stuck'
+
+opaque type TransactionRecord = {|
+  id: TransactionId,
+  hash: Hash,
+  status: TransactionState,
+  asset: AssetAddress,
+  from: Address, // FIXME: Contact: Address | OwnerAddress | ContactID
+  to: Address,
+  fee: string,
+  note: string,
+  timestamp: number,
+|}
+
+export type TransferIn = {|
+  ...TransactionRecord,
+  +type: typeof TRANSFER_IN_TYPE,
+  amount: string,
+|}
+
+export type TransferOut = {|
+  ...TransactionRecord,
+  +type: typeof TRANSFER_OUT_TYPE,
+  amount: string,
+|}
+
+export type TransferCancel = {|
+  ...TransactionRecord,
+  +type: typeof TRANSFER_CANCEL_TYPE,
+|}
+
+export type ContractCall = {|
+  +type: typeof CONTRACT_CALL_TYPE,
+|}
+
+export type EventMint = {|
+  +type: typeof EVENT_MINT_TYPE,
+|}
+
+export type EventBurn = {|
+  +type: typeof EVENT_BURN_TYPE,
+|}
+
+export type HistoryItem
+  = TransferIn
+  | TransferOut
+  | TransferCancel
+  | ContractCall
+  | EventMint
+  | EventBurn
```

Comment:
I would suggest to move this file into `flow-typed/redux/history`, and `declare` many of these types.

Reply:
Global declarations of types is bad practice, because  they don't declare a context of using